adal/src/main/java/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -437,12 +437,8 @@ public void acquireToken(Activity activity, String resource, String clientId,
     }
 
     /**
-     * acquireToken will start interactive flow if needed. It checks the cache
-     * to return existing result if not expired. It tries to use refresh token
-     * if available. If it fails to get token with refresh token, behavior will
-     * depend on options. If promptbehavior is AUTO, it will remove this refresh
-     * token from cache and fall back on the UI if activitycontext is not null.
-     * Default is AUTO.
+     * acquireToken will start an interactive auth flow to acquire new tokens 
+     * with the requested claims. Bypasses token cache. 
```

Comment:
Was there some specific ask to change this documentation? These calls will only bypass the token cache if `PromptBehavior != Auto` _or_ claims challenges are present.

[Here ](https://github.com/AzureAD/azure-activedirectory-library-for-android/blob/dev/adal/src/main/java/com/microsoft/aad/adal/AcquireTokenRequest.java#L333-L341)is where this call uses the local cache to try and acquireTokens -- [here ](https://github.com/AzureAD/azure-activedirectory-library-for-android/blob/dev/adal/src/main/java/com/microsoft/aad/adal/AcquireTokenRequest.java#L386-L390)is where claims challenges and `PrompBehavior` are evaluated

Reply:
Good catch. The code here may be implemented a bit differently than the other platforms and planned behavior ([ADAL .NET](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/blob/dev/adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/AuthenticationContext.cs#L391)).  I'm not sure we need to change it, but it does seem confusing to have split behavior in a call intended for getting a token with a superset of claims. IMO, claims in this overload should be non-optional.

Reply:
Submitted a new commit with correct behavior. The change is coming from some confusion/ambiguity around this method.